From 871904814bc8fca82f375d96152b8f05cb20fafb Mon Sep 17 00:00:00 2001
From: Geetha sowjanya <gakula@marvell.com>
Date: Mon, 1 Apr 2019 17:33:49 +0530
Subject: [PATCH 129/386] octeontx2-af: Fix rvu probe on cgx disable

RVU probe fails when no cgx device are connected.
This patch fix the issue.

Change-Id: I92e3c68dbca1b8f21e4ffda077cf37fcc6a12714
Signed-off-by: Geetha sowjanya <gakula@marvell.com>
Reviewed-on: https://sj1git1.cavium.com/6694
Tested-by: sa_ip-sw-jenkins
Reviewed-by: Sunil Kovvuri Goutham <Sunil.Goutham@cavium.com>
[RH: Original patch taken from marvell 88F3720 board support SDK 10.0-PR2003]
Signed-off-by: Ruiqiang Hao <Ruiqiang.Hao@windriver.com>
---
 drivers/net/ethernet/marvell/octeontx2/af/cgx.c | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/drivers/net/ethernet/marvell/octeontx2/af/cgx.c b/drivers/net/ethernet/marvell/octeontx2/af/cgx.c
index abd2012404e9..98dd6b1dc575 100644
--- a/drivers/net/ethernet/marvell/octeontx2/af/cgx.c
+++ b/drivers/net/ethernet/marvell/octeontx2/af/cgx.c
@@ -731,7 +731,7 @@ int cgx_get_mkex_prfl_info(u64 *addr, u64 *size)
 	if (!addr || !size)
 		return -EINVAL;
 
-	cgx_dev = list_first_entry(&cgx_list, struct cgx, cgx_list);
+	cgx_dev = list_first_entry_or_null(&cgx_list, struct cgx, cgx_list);
 	if (!cgx_dev)
 		return -ENXIO;
 
@@ -838,7 +838,7 @@ int cgx_get_fwdata_base(u64 *base)
 	struct cgx *cgx;
 	int err;
 
-	cgx = list_first_entry(&cgx_list, struct cgx, cgx_list);
+	cgx = list_first_entry_or_null(&cgx_list, struct cgx, cgx_list);
 	if (!cgx)
 		return -ENXIO;
 
-- 
2.17.1

